universal-application-tool-0.0.1/app/services/question/QuestionDefinition.java

```diff
@@ -5,12 +5,23 @@
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
 import com.google.common.collect.ImmutableMap;
 import java.util.Locale;
 import java.util.Optional;
 
 /** Defines a single question. */
-public class QuestionDefinition {
+@JsonTypeInfo(
+    use = JsonTypeInfo.Id.NAME,
+    include = JsonTypeInfo.As.PROPERTY,
+    property = "serialization_type")
+@JsonSubTypes({
+  @JsonSubTypes.Type(value = AddressQuestionDefinition.class, name = "address"),
+  @JsonSubTypes.Type(value = NameQuestionDefinition.class, name = "name"),
+  @JsonSubTypes.Type(value = TextQuestionDefinition.class, name = "text")
+})
+public abstract class QuestionDefinition {
```

Comment:
suggestion (blocking): `getQuestionType` and `getScalars` below should be abstract methods that are implemented by the subclasses